src/Automation/Automation/Common/AutomationPSClient.cs

```diff
@@ -1210,8 +1210,15 @@ public void DeleteCertificate(string resourceGroupName, string automationAccount
 
             if (connectionModel.FieldDefinitionValues.ContainsKey(connectionFieldName))
             {
-                connectionModel.FieldDefinitionValues[connectionFieldName] =
-                    PowerShellJsonConverter.Serialize(value);
+                if (value.GetType().FullName == "System.String")
```

Comment:
@ikanni any reason that we can't change this check to the following?

```cs
if (value is string)
{
    ...
}
```

Reply:
Thank you checking @cormacpayne